iap_mail_domain_blacklist/hooks.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 Quartile
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.sql_db import db_connect
+from odoo.tools import config
+
+from odoo.addons.iap.tools import iap_tools
+
+
+def update_mail_domain_blacklist(cr):
+    cr.execute(
+        "SELECT value FROM ir_config_parameter WHERE key = %s",
+        ("iap_mail_domain_blacklist.mail_domain_blacklist",),
+    )
+    param_value = cr.fetchone()
+    if param_value and param_value[0]:
+        additional_domains = {domain.strip() for domain in param_value[0].split(",")}
+        iap_tools._MAIL_DOMAIN_BLACKLIST.update(additional_domains)
+
+
+def post_load_hook():
+    db_name = config["db_name"]
+    if db_name:
```

Comment:
With `from odoo.service import db`. I don't think we can rely on the `db_name` param, although I'm not sure this is the best approach.
```suggestion
    for db_name in db.list_dbs():
```